sklearn/svm/_classes.py

```diff
@@ -515,6 +545,24 @@ def fit(self, X, y, sample_weight=None):
             accept_large_sparse=False,
         )
         penalty = "l2"  # SVR only accepts l2 penalty
+
+        if self.dual == "auto":
+            if self.loss == "epsilon_insensitive":
+                self._dual = True
+            else:
+                self._dual = True if X.shape[0] < X.shape[1] else False
```

Comment:
I would reuse the same code than above.

Actually, I think that we could make this `dual` validation inside a function that could be called by both the regressor and classifier.